Feat: система регистрации

Roman821 opened the pull request:
Тык-тык, клац-клац ⌨️
Пара бессонных ночей ✨
Дядя Никита проверит и закинет в ветку this изменения 💅

Roman821 committed:
Add files via upload

Roman821 committed:
feat: система регистрации

routers/users.py:
	Создание пользователя (username, password, discord_id).
	Проверка try, except на уникальность username или discord id.

sql/utils.py:
	В def create_superuser добавили discord id
	+ Проверяем, что на вход дали достаточное количество аргументов (5, а не 4)

core/config.py:
	Добавили в словарь OAUTH2_SCHEME_SCOPES область "register" которая предоставляет api право регистрировать новых пользователей.

core/schemas.py:
	В классе UserBase объявляем новый атрибут discord_id

Roman821 committed:
feat: обновление команды

Добавили в команду аргумент "discord_id"

LaGGgggg commented on API/routers/users.py:
```diff
@@ -74,3 +75,41 @@ def get_user_me_data(
+
+    # Создание нового пользователя в базе данных
+
+    # return new_user
```
Это что такое? Мёртвый код? Серьёзно?

Roman821 replied:
Фикс-фикс *magic*
Всё исправили 👌

LaGGgggg commented on API/routers/users.py:
```diff
@@ -74,3 +75,41 @@ def get_user_me_data(
+    try:
+        return schemas.User.from_orm(crud.create_user(db, schemas.UserCreate(
+            username=username, password=password, discord_id=discord_id
+        ), pwd_context))
```
А если написать так:
```python
return schemas.User.from_orm(crud.create_user(
    db, schemas.UserCreate(username=username, password=password, discord_id=discord_id), pwd_context
))
```
Не будет ли лучше выглядеть? Звучит как вкусовщина, поэтому это не претензия, а предложение. Однако, мне кажется что в моём варианте чуть понятнее выглядит за счёт `pwd_context`, который не висит непонятно где

Roman821 replied:
Теперь выглядит лучше)) 😸

LaGGgggg commented on API/routers/users.py:
```diff
@@ -74,3 +75,41 @@ def get_user_me_data(
+        password: str,
+        discord_id: int,
+        db: Session = Depends(get_db),
+        current_user: models.User = Security(login_manager, scopes=['register']),
```
Думаю, что `current_user` лучше заменить на `_`, так как мы этот аргумент в функции не используем (но удалять мы его не будем, так как он нужен для валидации scopes пользователя силами FastAPI)
P.S. Это где-то в PEP-8, мне кажется, прописано, для неиспользуемых аргументов/переменных используется `_` в качестве названия

Roman821 replied:
Замениль 🌵

LaGGgggg reviewed (changes requested):
В целом всё ок, пункты из ТЗ задачи выполнены, всё работает, молодец!
***(но всё равно по вопросикам из review пройдись)***
Ах да, коммиты, мне не очень понравилось как они написаны. Во-первых, они оставляют вопросы относительно того, что именно поменялось. Во-вторых, там есть фактическая ошибка. Но, так как это первый опыт, я особо не докапываюсь.
Всё остальное мне понравилось

Roman821 committed:
feat: улучшение читабельности

В функции create_user заменил "_" вместо "current_user"  (означает, что мы не собираемся использовать значение этой переменной)
Удалил "мертвый код"
Улучшил читабельность кода

alinkalina reviewed (approved):
Откровенно говоря, не стала сильно вдаваться в подробности, что конкретно тут происходит, но судя по тому, что это **Система регистрации** видимо что-то важное)) Ну а вообще по-моему **всё круто**!
